Info-Orbs/src/widgets/stockScreenWidget.cpp

```diff
@@ -0,0 +1,113 @@
+#include "widgets/stockScreenWidget.h"
+#include <config.h>
+#include <sstream>
+#include <iomanip>
+#include <HTTPClient.h>
+#include <ArduinoJson.h>
+
+StockScreenWidget::StockScreenWidget(ScreenManager &manager) : Widget(manager), currentStockIndex(0), m_stockDelay(6000) {
+}
+
+void StockScreenWidget::setup() {
+    m_stockDelayPrev = 0;
+    stockTickers = {"ABNB", "BRCC", "GOOG", "TSLA", "GME"};
+}
+
+void StockScreenWidget::draw() {
+    // Get initial stock then every m_stockDelay milliseconds
+    if (m_stockDelayPrev == 0 || (millis() - m_stockDelayPrev) >= m_stockDelay) {
+        getStockData();
+        displayStock(currentStockIndex, TFT_WHITE, TFT_BLACK);
+        m_stockDelayPrev = millis();
+        currentStockIndex = (currentStockIndex + 1) % stockTickers.size();
+    }
+}
+
+void StockScreenWidget::update() {
+}
+
+void StockScreenWidget::getStockData() {
```

Comment:
what if you put this all into the update() method and had everything store to an array so that you keep the data for the next time it is displayed? This is what the weatherWidget is now doing. And it allows cycling through widgets without forcing new data every time if the data is still current.

Reply:
That sounds good. A little more advance for me at the moment.